graviti/manager/commit.py

```diff
@@ -191,58 +189,55 @@ class CommitManager:
     """This class defines the operations on the commit in Graviti.
 
     Arguments:
-        dataset: :class:`~graviti.dataset.dataset.Dataset` instance.
+        access_info: :class:`~graviti.manager.dataset.DatasetAccessInfo` instance.
+        commit_id: The commit id.
 
     """
 
-    def __init__(self, dataset: "Dataset") -> None:
-        self._dataset = dataset
+    def __init__(self, access_info: "DatasetAccessInfo", commit_id: str) -> None:
+        self._access_info = access_info
+        self._commit_id = commit_id
 
     def _generate(
         self, revision: Optional[str], offset: int = 0, limit: int = 128
-    ) -> Generator[TensorbayCommit, None, int]:
+    ) -> Generator[Commit, None, int]:
         if revision is None:
-            revision = self._dataset.commit_id
-
+            revision = self._commit_id
         if not revision:
             raise TypeError("The given revision is illegal")
 
         response = list_commits(
-            self._dataset.url,
-            self._dataset.access_key,
-            self._dataset._dataset_id,  # pylint: disable=protected-access
-            commit=revision,
+            **self._access_info,
+            revision=revision,
             offset=offset,
             limit=limit,
         )
 
         for item in response["commits"]:
-            yield TensorbayCommit.loads(item)
+            yield Commit.from_pyobj(item)
 
         return response["totalCount"]  # type: ignore[no-any-return]
 
-    def get(self, revision: Optional[str] = None) -> TensorbayCommit:
+    def get(self, revision: Optional[str] = None) -> Commit:
         """Get the certain commit with the given revision.
 
         Arguments:
             revision: The information to locate the specific commit, which can be the commit id,
                 the branch name, or the tag name. If it is not given, get the current commit.
 
-        Raises:
-            ResourceNotExistError: When the required commit does not exist.
-
         Returns:
-            The :class:`.TensorbayCommit` instance with the given revision.
+            The :class:`.Commit` instance with the given revision.
 
         """
-        try:
-            commit = next(self._generate(revision))
-        except StopIteration as error:
-            raise ResourceNotExistError(resource="commit", identification=revision) from error
+        if revision is None:
+            commit_id = self._commit_id
+        else:
+            commit_id = get_revision(**self._access_info, revision=revision)["commit_id"]
```

Comment:
Should `get_revision` returns all commit info? if yes, why we need to call `get_commit`?

Reply:
`get_revision` returns all commit info. we do not need to call get_commit here.  So we do not need api `get commit`